src/object.h

```diff
@@ -0,0 +1,195 @@
+
+/* Redis object implementation */
+void decrRefCount(robj *o);
+void incrRefCount(robj *o);
+robj *makeObjectShared(robj *o);
+void freeStringObject(robj *o);
+void freeListObject(robj *o);
+void freeSetObject(robj *o);
+void freeZsetObject(robj *o);
+void freeHashObject(robj *o);
+void dismissObject(robj *o, size_t dump_size);
+robj *createObject(int type, void *ptr);
+void initObjectLRUOrLFU(robj *o);
+robj *createStringObject(const char *ptr, size_t len);
+robj *createRawStringObject(const char *ptr, size_t len);
+robj *tryCreateRawStringObject(const char *ptr, size_t len);
+robj *tryCreateStringObject(const char *ptr, size_t len);
+robj *dupStringObject(const robj *o);
+int isSdsRepresentableAsLongLong(sds s, long long *llval);
+int isObjectRepresentableAsLongLong(robj *o, long long *llongval);
+robj *tryObjectEncoding(robj *o);
+robj *tryObjectEncodingEx(robj *o, int try_trim);
+size_t getObjectLength(robj *o);
+robj *getDecodedObject(robj *o);
+size_t stringObjectLen(robj *o);
+robj *createStringObjectFromLongLong(long long value);
+robj *createStringObjectFromLongLongForValue(long long value);
+robj *createStringObjectFromLongLongWithSds(long long value);
+robj *createStringObjectFromLongDouble(long double value, int humanfriendly);
+robj *createQuicklistObject(int fill, int compress);
+robj *createListListpackObject(void);
+robj *createSetObject(void);
+robj *createIntsetObject(void);
+robj *createSetListpackObject(void);
+robj *createHashObject(void);
+robj *createZsetObject(void);
+robj *createZsetListpackObject(void);
+robj *createStreamObject(void);
+robj *createModuleObject(struct RedisModuleType *mt, void *value);
+int getLongFromObjectOrReply(struct client *c, robj *o, long *target, const char *msg);
+int getPositiveLongFromObjectOrReply(struct client *c, robj *o, long *target, const char *msg);
+int getRangeLongFromObjectOrReply(struct client *c, robj *o, long min, long max, long *target, const char *msg);
+int checkType(struct client *c, robj *o, int type);
+int getLongLongFromObjectOrReply(struct client *c, robj *o, long long *target, const char *msg);
+int getDoubleFromObjectOrReply(struct client *c, robj *o, double *target, const char *msg);
+int getDoubleFromObject(const robj *o, double *target);
+int getLongLongFromObject(robj *o, long long *target);
+int getLongDoubleFromObject(robj *o, long double *target);
+int getLongDoubleFromObjectOrReply(struct client *c, robj *o, long double *target, const char *msg);
+int getIntFromObjectOrReply(struct client *c, robj *o, int *target, const char *msg);
+char *strEncoding(int encoding);
+int compareStringObjects(const robj *a, const robj *b);
+int collateStringObjects(const robj *a, const robj *b);
+int equalStringObjects(robj *a, robj *b);
+void trimStringObjectIfNeeded(robj *o, int trim_small_values);
+
+void dismissSds(sds s);
+struct redisMemOverhead *getMemoryOverheadData(void);
+void freeMemoryOverheadData(struct redisMemOverhead *mh);
```

Comment:
these methods doesn't relate to robj or kvobj, maybe keep them in the original place? 
similar as objectCommand() and memoryCommand().